### ISSUE TITLE
CSS is not updated when value of a variable changes

### ISSUE THREAD

**Issue by brijeshb42** [callstack/linaria#1265]
## Environment

- Linaria version: latest
- Bundler (+ version): vite@4.3.2
- Node.js version:18.11.0
- OS: MacOS (M1 Max)

## Description

CSS content is not updated in HMR when value of any variable that is being used in the tagged template literal changes.

## Reproducible Demo

https://codesandbox.io/p/sandbox/zen-wescoff-xt5289?file=%2Fsrc%2FApp.tsx%3A41%2C18

Just change value of the `fs` variable.

https://github.com/callstack/linaria/assets/717550/a3f29117-dbcf-4377-bee6-4ea7dcdc382d

**Comment by ShayDavidson** [comment]
We have the same problem as well. Not only that it doesn't refresh, it requires us to restart Vite in order for it to update!

Our example is this:
```
const testStyle = css`
  font-size: 50px;
  color: ${themedColor('primary')};
`;
```

Where `themedColor` is just a function helper that constructs a namespaced variable name. Changing the argument from `primary` to something else doesn't work - they value is stuck as `primary` until I restart Vite.

If I do this however, outside of the `css` scope:
```
export const App: React.FC = () => {
  return (
    <div style={{ color: themedColor('primary') }}>Hello</div>
  );
};
```

And change the value there, the HMR works well.

**Comment by Anber** [comment]
@ShayDavidson the original issue was fixed in #1285. Could you please check if it also fixes your case?

**Comment by ShayDavidson** [comment]
@Anber thank you, it did work!
